carbon-design-system/carbon-for-ibm-dotcom
# fix(card-in-card-5984): default to using Card component headline styling

### Related Ticket(s)

GH Issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5984
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2131

### Description

- This PR removes unneeded styling defined in the `scss` partial of `card-in-card`.
- Card in Card headline should now be inheritancing the styling defined in the `card/index.scss`(see [link](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/packages/styles/scss/components/card/index.scss#L416)).

### Testing Instruction

- Visit the `Card Group with Card in Card` - `/?path=/story/components-card-group--with-card-in-card`
- Ensure that the headline within the `<dds-card-in-card>` component is inheritancing the styling defined in the `card/index.scss`(see [link](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/packages/styles/scss/components/card/index.scss#L416)) or `expressive-heading-03` responsive mode.

### Expected Display

![Screen Shot 2021-08-10 at 4 40 58 PM](https://user-images.githubusercontent.com/1815714/128932085-9d464f47-e469-4095-9793-5f2a3500e8fb.png)
